content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+### Collections
+**Collections** according to C# are classes that hold together groups of objects. The `.NET` libraries provide useful system-defined classes, like collection namespace, which provide various collections and their functionalities. The use of non-generic collections is normally discouraged therefore my article fully focuses on generic collections.
+
+We have three common generic collections incorporated in the C# namespace.
+
+## 1. LIST
+
+A list is the most commonly used generic collection. Its because they are much more convenient when they are used instead of an array.
+A list is a very general data structure with each entry occupying a specific position known as an index. Lists are very flexible as they automatically resize every time data is added.
+
+Lists are very easy to manipulate. This is because many methods have been created and abstracted, meaning the bulky work is already done for the user. The user only has to call the method and manipulate the list.
+ 
+A list stores data in sequence, although the sequence oughtn't to have to be particular. It can also accept duplicate entries.
+Below is a pictorial explanation of a list. A generic list class is declared as shown:
+
+```
+public class genericList<T>
+
+{
+    
+    void add(T input){}
+}
+```
+
+On the same note, a list can be of integer or string data type. They are declared differently but the syntax is the same as shown:
+```
+using system;
+
+{
+private class genericListExample
+static void Main ()
+{
+    genericlist<int> list1 = new genericlist<int>(); //This declares a  list of type int.
+    
+     genericlist<string> list2 = new genericlist<string>(); //This declares a list of type string.
+     
+      genericlist<int> list1 = new genericlist<int>(); //This declares a list of type int.
+}
+}
+```
+
+A simple c# example of generic list implementation might help in understanding.
+
+```
+using system;
+using system.collection.generic; // A system namespace for implementing generic lists.
+
+namespace GenericListExample
+{
+    public class Generic<T>  // Declare Generics.
+    {
+        public void GenFunction(T printvalue)
+        {
+            Console.WriteLine(printvalue)
+        }
+    }
+    public class program
+    {
+        public static void Main(string [] args)
+        {
+            Console.WriteLine("Printing integer value");
+            Generic<int> gen = new Generic<int>();
+            gen.GenFunction(10);
+            
+            Console.WriteLine("Printing string value");
+            Generic<string> genString = new Generic<string>();
+            gen.GenFunction("Hello world");
+            
+        }
+    }
+}
+```
+
+Note that _<T>_ is a placeholder for data types and replaced by datatype at compile time.
```

Comment:
> Note that _<T>_ is a placeholder for data types and replaced by data type at compile time

There is a bit of a formatting issue here. In markdown, *\>* indicates that the following paragraph is a blockquote. You need to escape the *\>* by placing a *\\* in front like so: 

> _<T\\>_ 

This way it look like this:

> _<T\>_

instead of:

> _<T>_

(doesn't even appear). 

Also: replace the *_* 's with *\**'s

Reply:
> Note that _<T>_ is a placeholder for data types and replaced by datatype at compile time.

Change the wording to:

> Note that *<T\>* is a placeholder that is replaced by a data type at compile time. 

To be more grammatically correct and concise.